common/core/desktop/tests/unit/kmx/kmx.cpp

```diff
@@ -168,8 +171,8 @@ void apply_action(km_kbp_state const *, km_kbp_action_item const & act, std::u16
     break;
   case KM_KBP_IT_CHAR:
     if (Uni_IsSMP(act.character)) {
-      text_store.push_back(Uni_IsSurrogate1(act.character));
-      text_store.push_back(Uni_IsSurrogate2(act.character));
+      text_store.push_back(Uni_UTF32ToSurrogate1(act.character));
+      text_store.push_back(Uni_UTF32ToSurrogate2(act.character));
```

Comment:
Another lovely bug. We were pushing booleans back onto `text_store` in the test runner. This issue was never spotted, because we were never receiving the expected U+10000+ values, due to the bug in kmx_processor.cpp:425!

Reply:
🔍